optuna/samplers/_partial_fixed.py

```diff
@@ -0,0 +1,115 @@
+from typing import Any
+from typing import Dict
+import warnings
+
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+
+
+@experimental("2.3.0")
+class PartialFixedSampler(BaseSampler):
+    """Sampler that can sample parameters with fixed parameters.
+
+        .. versionadded:: 2.3.0
+
+    Example:
+
+         After several steps of optimization, you can fix the value of ``y`` and re-optimize it.
+
+        .. testcode::
+
+            import optuna
+
+
+            def objective(trial):
+                x = trial.suggest_uniform("x", -1, 1)
+                y = trial.suggest_int("y", -1, 1)
+                return x ** 2 + y
+
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=10)
+
+            best_params = study.best_params
+            fixed_params = {"y": best_params["y"]}
+            partial_sampler = optuna.samplers.PartialFixedSampler(fixed_params, study.sampler)
+
+            study.sampler = partial_sampler
+            study.optimize(objective, n_trials=10)
+
+    """
+
+    def __init__(self, fixed_params: Dict[str, Any], base_sampler: BaseSampler) -> None:
+        self._fixed_params = fixed_params
+        self._base_sampler = base_sampler
+
+    def reseed_rng(self) -> None:
+
+        self._rng = self._base_sampler.reseed_rng()  # type: ignore
```

Comment:
I think we just need to call`self._base_sampler.reseed_rng()` here because `reseed_rng()` returns `None`. As a result, we can drop `# type: ignore`.

Reply:
Thank you so much!